### ISSUE TITLE
[ ubuntu ]  [ 3-D ] GLUT can run well but not for openpose ,  undefined reference to ‘gluPerspective’ 

### ISSUE THREAD

**Issue by Qidian213** [CMU-Perceptual-Computing-Lab/openpose#666]
# Type of Issue

- Compilation/installation error
   undefined reference to ‘gluPerspective’
   
# Your System Configuration
1.OpenPose version : openpose 1.3.0 , ( update 7 days ago )
     
2. General configuration
    -Installation mode: CMake

    -Operating system : ubuntu 16.04 
    
    -Release or Debug mode?: release 
    
    -Compiler : gcc 5.4.0 (Ubuntu)

3. 3-D Reconstruction module added ? : yes 
    
4. 3rd-party software:
     -Caffe version: Default 

    - CMake version: cmake 3.11.0
    
    - OpenCV version: compiled from source  opencv3.2.0
    
5. GPU mode:
    - CUDA version: cuda8
    
    - cuDNN version : cudnn5.1.10
    
    - GPU model : GTX1060 , NVIDIA-SMI 384.130

# I installed freeglut by :

  sudo apt-get update
  
  sudo apt-get install build-essential
  
  sudo apt-get install freeglut3 freeglut3-dev libxmu-dev libxi-dev

files can found on my system:

   /usr/include/GL/freeglut.h
   /usr/include/GL/freeglut_std.h
   /usr/include/GL/glut.h
  /usr/include/GL/freeglut_ext.h
  /usr/lib/x86_64-linux-gnu/libglut.so.3
  /usr/lib/x86_64-linux-gnu/libglut.so
  /usr/lib/x86_64-linux-gnu/libglut.a
  /usr/lib/x86_64-linux-gnu/libglut.so.3.9.0
  
when using CMake-gui, check the "Advanced" checkbox , cmake can found  them same as above .

# Issue Summary

   when i use test code test.c ,it can run well .

   compile cmd :   gcc test.c -lGL -lGLU -lglut -o test  

   `#include <GL/glut.h>
    void init(void)
{
        glClearColor(0.0, 0.0, 0.0, 0.0);
        glMatrixMode(GL_PROJECTION);
        glOrtho(-5, 5, -5, 5, 5, 15);
        glMatrixMode(GL_MODELVIEW);
        gluLookAt(0, 0, 10, 0, 0, 0, 0, 1, 0);
        return;
}

void display(void)
{
        glClear(GL_COLOR_BUFFER_BIT);
        glColor3f(1.0, 0, 0);
        glutWireTeapot(3);
        glFlush();
        return ;
}

int main(int argc, char *argv[])
{
        glutInit(&argc, argv);
        glutInitDisplayMode(GLUT_RGB | GLUT_SINGLE);
        glutInitWindowPosition(0, 0);
        glutInitWindowSize(800, 800);
        glutCreateWindow("OpenGL 3D View");
        init();
        gluPerspective (0,0,0,0);
       glutDisplayFunc(display);
       glutMainLoop();
       return 0;
}`

but when i compile openpose with WITH_3D_RENDERER ,  error occur : 

   `[ 87%] Linking CXX shared library libcamera.so
[ 87%] Built target camera
Scanning dependencies of target calibration.bin
[ 88%] Building CXX object examples/calibration/CMakeFiles/calibration.bin.dir/calibration.cpp.o
[ 88%] Linking CXX executable calibration.bin
../../src/openpose/libopenpose.so.1.3.0：Undefined reference to ‘gluPerspective’
../../src/openpose/libopenpose.so.1.3.0：Undefined reference to 'gluLookAt'
collect2: error: ld returned 1 exit status
examples/calibration/CMakeFiles/calibration.bin.dir/build.make:162: recipe for target 'examples/calibration/calibration.bin' failed
make[2]: *** [examples/calibration/calibration.bin] Error 1
CMakeFiles/Makefile2:1210: recipe for target 'examples/calibration/CMakeFiles/calibration.bin.dir/all' failed
make[1]: *** [examples/calibration/CMakeFiles/calibration.bin.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
`

**Comment by gineshidalgo99** [comment]
Doc clearly indicates the dependencies in:
https://github.com/CMU-Perceptual-Computing-Lab/openpose/blob/master/doc/installation.md#3d-reconstruction-module

INstruction:
```
sudo apt-get update && sudo apt-get install build-essential freeglut3 freeglut3-dev libxmu-dev libxi-dev
```

**Comment by Qidian213** [comment]
Thanks , i found many issues talk about it  and i had try the instruction many times and reboot my PC , but no use for me . I use the test code to test  GLUT , it can work well , but not in OPENPOSE ,

**Comment by sergionegri** [comment]
We are running into the same issue: GLUT working standalone but not with Openpose (same compilation error of undefined reference to gluPerspective).

We have checked the presence of the following files in their corresponding directory (as indicated in the CMakeGUI:

```
GLUT_Xi_LIBRARY: /usr/lib/x86_64-linux-gnu/libXi.so
GLUT_Xmu_LIBRARY: /usr/lib/x86_64-linux-gnu/libXmu.so
GLUT_glut_LIBRARY: /usr/lib/x86_64-linux-gnu/libglut.so
```

And they are present.

`GLUT_INCLUDE_DIR is the same as any other library (/usr/include).`

@Qidian213 did you have any luck?

**Comment by Qidian213** [comment]
@sergionegri no, i give up

**Comment by sergionegri** [comment]
@Qidian213 we found the solution. Have a look here at my answer ;) https://github.com/CMU-Perceptual-Computing-Lab/openpose/issues/894
Hope this helps

**Comment by gineshidalgo99** [comment]
Issue fixed after PR #771
